src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,435 @@
+[<NoComparison; NoEquality; RequireQualifiedAccess>]
+type CachingStrategy =
+    /// Retain a single 'state per streamName.
+    /// Each cache hit for a stream renews the retention period for the defined <c>window</c>.
+    /// Upon expiration of the defined <c>window</c> from the point at which the cache was entry was last used, a full reload is triggered.
+    /// Unless <c>LoadOption.AllowStale</c> is used, each cache hit still incurs a roundtrip to load any subsequently-added events.
+    | SlidingWindow of ICache * window : TimeSpan
+    /// Retain a single 'state per streamName
+    /// Upon expiration of the defined <c>period</c>, a full reload is triggered.
+    /// Unless <c>LoadOption.AllowStale</c> is used, each cache hit still incurs a roundtrip to load any subsequently-added events.
+    | FixedTimeSpan of ICache * period : TimeSpan
+    /// Prefix is used to segregate multiple folds per stream when they are stored in the cache.
+    /// Semantics are identical to <c>SlidingWindow</c>.
+    | SlidingWindowPrefixed of ICache * window : TimeSpan * prefix : string
+
+type MessageDbCategory<'event, 'state, 'context>(resolveInner, empty) =
+    inherit Equinox.Category<'event, 'state, 'context>(resolveInner, empty)
+    new (   context : MessageDbContext, codec : IEventCodec<_, _, 'context>, fold, initial,
+            [<O; D(null)>]?caching,
+            [<O; D(null)>]?access) =
+        let inner = Category<'event, 'state, 'context>(context, codec, ?access = access)
+        let readCacheOption =
+            match caching with
+            | None -> None
+            | Some (CachingStrategy.SlidingWindow (cache, _))
+            | Some (CachingStrategy.FixedTimeSpan (cache, _)) -> Some (cache, null)
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, _, prefix)) -> Some (cache, prefix)
+        let folder = Folder<'event, 'state, 'context>(inner, fold, initial, ?readCache = readCacheOption)
+        let category : ICategory<_, _, 'context> =
+            match caching with
+            | None -> folder :> _
+            | Some (CachingStrategy.SlidingWindow (cache, window)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache null window folder Token.supersedes
+            | Some (CachingStrategy.FixedTimeSpan (cache, period)) ->
+                Caching.applyCacheUpdatesWithFixedTimeSpan cache null period folder Token.supersedes
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, window, prefix)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache prefix window folder Token.supersedes
+        let resolveInner streamIds = struct (category, StreamName.Internal.ofCategoryAndStreamId streamIds, ValueNone)
+        let empty = struct (context.TokenEmpty, initial)
+        MessageDbCategory(resolveInner, empty)
+
+
+type MessageDbConnector(
+    connectionString : string, ?readOnlyConnectionString : string,
```

Comment:
the `Only` is not talking to me (and if it was, we should apply it to similar cases in other stores) ?
not a deal breaker...
also needs `O;D null` (dont think there are any more...)